### PR TITLE
Upgrade to match core 2.0

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
closing #169 
closing #196 
closing #206 
closing #207 

U2F support and enroll token function - removed

### CODE REVIEW

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: composer.json [hunk 1]
```diff
@@ -1,12 +1,19 @@
 {
     "name": "privacyidea/simplesamlphp-module-privacyidea",
-    "description": "Add two factor authentication like OTP to simpleSAMLphp. The second factor is managed in privacyIDEA.",
+    "description": "Add Multi-Factor Authentication to simpleSAMLphp. The MFA is managed in privacyIDEA.",
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply composer.json:3]: second factors

[the remark this answers is not in the document]
Comment (nilsbehlen) [reply composer.json:3]: Add multi-factor authentication to simpleSAMLphp. The secind factors are managed in privacyIDEA.

[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply composer.json]: [8b8eda4](https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/209/commits/8b8eda4f9bd3a17e0113f0f4bd19339a07575d06)

File: README.md [hunk 2]
```diff
@@ -22,5 +19,12 @@ Please check the [documentation](https://github.com/privacyidea/simplesamlphp-mo
...
 
-## Logfiles
-The saml log can be read with `journalctl -f`. If you encounter any problems that are not logged as errors, set the logging level of simpleSAMLphp to debug by editing `{samlDir}/config/config.php`. Search for `logging.level` and set it so `SimpleSAML\Logger::DEBUG`. Alternatively, the apache error log can be checked for the errors. It is located at `/var/log/apache2/error.log`.
+## Logging
+The saml log can be read with `journalctl -f`. 
+If you encounter any problems that are not logged as errors, 
+set the logging level of simpleSAMLphp to debug by 
+editing `{samlDir}/config/config.php`. 
+Search for `logging.level` and set it 
+so `SimpleSAML\Logger::DEBUG`. Alternatively,
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply README.md]: [079ae04](https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/209/commits/079ae0476e07d1bd8fe66b3af40c3218b46cfcc6)

File: src/Auth/Process/PrivacyideaAuthProc.php [hunk 3]
```diff
@@ -26,17 +25,17 @@
 class PrivacyideaAuthProc extends ProcessingFilter
 {
     /* @var array This contains the authproc configuration which is set in metadata */
-    private $authProcConfig;
-    /* @var PrivacyIDEA This is an object from privacyIDEA class */
-    private $pi;
+    private array $authProcConfig;
+    /* @var PrivacyIDEA|null This is an object from privacyIDEA class */
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply src/Auth/Process/PrivacyideaAuthProc.php]: [f6d2133](https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/209/commits/f6d213399ebf0dee7c26a081461f70f703359a13)

File: src/Auth/PrivacyIDEA.php [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+     * @throws PIBadRequestException If an error occurs during the request.
+     */
+    public function getAuthToken(): string
+    {
+        if (!$this->serviceAccountAvailable())
+        {
+            $this->errorLog("Cannot retrieve auth token without service account!");
+            return "";
+        }
+
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply src/Auth/PrivacyIDEA.php:341]: [f6d2133](https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/209/commits/f6d213399ebf0dee7c26a081461f70f703359a13)

File: public/FormBuilder.php [hunk 5]
```diff
@@ -49,7 +49,8 @@
 // Prepare the form to show
 try
 {
-    $tpl = new Template(Configuration::getInstance(), 'privacyidea:LoginForm.php');
+    //$tpl = new Template(Configuration::getInstance(), 'privacyidea:LoginForm.php');
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply public/FormBuilder.php]: [f6d2133](https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/209/commits/f6d213399ebf0dee7c26a081461f70f703359a13)

File: README.md [hunk 6]
```diff
@@ -1,14 +1,11 @@
 # privacyIDEA simpleSAMLphp Module
 
-This module adds flexible, enterprise-grade two-factor authentication 
+This module adds flexible, enterprise grade Multi-Factor Authentication
```
[the remark this answers is not in the document]
Comment (lukasmatusiewicz) [reply README.md]: [f6d2133](https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/209/commits/f6d213399ebf0dee7c26a081461f70f703359a13)

**Review by nilsbehlen — APPROVED** [review]

**Pull request merged**